api/src/main/java/org/pentaho/metaverse/api/analyzer/kettle/ExternalResourceCache.java

```diff
@@ -162,6 +163,18 @@ public String toString() {
   }
 
   public class TransValues extends Resources<Trans> {
+
+    @Override
+    public void add( final Trans value ) {
+      List<Trans> internalList = new ArrayList<>();
+      internalList.addAll( internal );
+      for ( Trans t : internalList ) {
```

Comment:
Why are you creating a new list? You could just iterate with the original internal list like in line 140
"for ( IExternalResourceInfo resource : internal ) {"